pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
and yeah, @deads2k summarized at least one of the scenarios the timeout was originally introduced to address.  proxy issues, and people specifying raw ips (but getting it wrong), or just everything was right except the git server was down, were others.